src/Kunstmaan/UtilitiesBundle/Helper/Shell/Shell.php

```diff
@@ -32,9 +36,14 @@ public function runInBackground($command, $priority = 0)
      */
     public function isRunning($pid)
```

Comment:
I see that the symfony Process object has a method called isRunning. Can you not use that method instead of this old convoluted way?

Reply:
```isRunning``` waiting for process to finish, but we need check output command ```ps -p PID -o pid```

Reply:
isRunning does not wait for a process to finish i think. That is what the Process->wait() function is for. The isRunning method just returns true or false.

Reply:
Tested on ```ls -l``` and always return false.

Reply:
ls -l is not a continuous process so there is no way the process could return true. Maybe try tail -f ~/.bashrc. This is not a process that will instantly quit.

Reply:
Ofcourse replace ~/.bashrc if you do not have that file.

Reply:
Also tested, it doesn't change anything.

Reply:
Ok if it does not work like that then we will use the existing way like you made it.